pkg/provisioningv2/rke2/planner/etcdrestore.go

```diff
@@ -61,8 +57,8 @@ func (p *Planner) runEtcdSnapshotRestorePlan(controlPlane *rkev1.RKEControlPlane
 
 	for _, server := range servers {
 		if isS3 ||
-			(server.Machine.Status.NodeRef != nil &&
-				server.Machine.Status.NodeRef.Name == snapshot.SnapshotFile.NodeName) {
+			(server.Machine.Labels[rke2.MachineIDLabel] != "" && snapshot.Labels[rke2.MachineIDLabel] != "" &&
```

Comment:
nit, only one check is enough: 
```suggestion
			(server.Machine.Labels[rke2.MachineIDLabel] != "" &&
```